src/map/entities/battleentity.cpp

```diff
@@ -256,10 +256,12 @@ int16 CBattleEntity::GetWeaponDelay(bool tp)
     if (!tp)
     {
         // Cap haste at appropriate levels.
-        int16 hasteMagic = std::clamp<int16>(getMod(Mod::HASTE_MAGIC), -448, 448);
-        int16 hasteAbility = std::clamp<int16>(getMod(Mod::HASTE_ABILITY), -256, 256);
-        int16 hasteGear = std::clamp<int16>(getMod(Mod::HASTE_GEAR), -256, 256);
-        WeaponDelay = (uint16)(WeaponDelay * ((1024.0f - hasteMagic - hasteAbility - hasteGear) / 1024.0f));
+        int16 hasteMagic = std::clamp<int16>(getMod(Mod::HASTE_MAGIC), -4375, 4375); // 43.75% cap
+        int16 hasteAbility = std::clamp<int16>(getMod(Mod::HASTE_ABILITY), -2500, 2500); // 25% cap
```

Comment:
https://github.com/DarkstarProject/darkstar/blob/master/src/map/utils/battleutils.cpp#L4253

was looking to do the type of thing we have going on with pdt

Reply:
@teschnei per convo in discord the thing pdt does may in fact be placebo, dumping this on you, I didn't check and I don't want to, I don't have time to find a years old commit and see exactly what happened along the way to present day.